pkg/karpenter/karpenter.go

```diff
@@ -0,0 +1,76 @@
+package karpenter
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kris-nova/logger"
+
+	"github.com/weaveworks/eksctl/pkg/karpenter/providers"
+)
+
+const (
+	// default namespace for Karpenter
+	KarpenterNamespace = "karpenter"
+
+	karpenterHelmRepo         = "https://charts.karpenter.sh"
+	karpenterHelmChartName    = "karpenter/karpenter"
+	karpenterReleaseName      = "karpenter"
+	controllerClusterName     = "controller.clusterName"
+	controllerClusterEndpoint = "controller.clusterEndpoint"
+	createServiceAccount      = "serviceAccount.create"
+	addDefaultProvisioner     = "defaultProvisioner.create"
+)
+
+// Options contains values which Karpenter uses to configure the installation.
+type Options struct {
+	HelmInstaller         providers.HelmInstaller
+	Namespace             string
+	ClusterName           string
+	AddDefaultProvisioner bool
+	CreateServiceAccount  bool
+	ClusterEndpoint       string
+	Version               string
+}
+
+// Manager defines a manager for Karpenter.
+type Manager interface {
+	InstallKarpenter(ctx context.Context) error
+	UninstallKarpenter(ctx context.Context) error
+}
+
+// Installer implements the Karpenter orchestrator using a HelmInstaller.
+type Installer struct {
+	Options
+}
+
+// NewKarpenterInstaller creates a new installer to configure and add Karpenter to a cluster.
+func NewKarpenterInstaller(opts Options) *Installer {
+	return &Installer{
+		Options: opts,
+	}
+}
+
+// InstallKarpenter adds Karpenter to a configured cluster in a separate CloudFormation stack.
+func (k *Installer) InstallKarpenter(ctx context.Context) error {
+	logger.Info("adding Karpenter to cluster %s with cluster endpoint", k.ClusterName, k.ClusterEndpoint)
```

Comment:
Nit: I'm not sure if logging the cluster endpoint is helpful to users.

Reply:
Hmm, hmmmm. Maybe it helps to see later on if it tried to ping it, and it didn't work. You could be able to easily see what it tried to access?

Reply:
We could also make it a debug log so that it doesn't appear in the default log level.

Reply:
Yep added that in another PR.